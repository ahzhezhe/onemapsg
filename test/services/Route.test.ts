import { oneMap } from '..';

test('route', async () => {
  const result = await oneMap.route.route({
    start: [1.3039454, 103.8408471],
    end: [1.3039454, 103.8408471],
    routeType: 'drive'
  });
  expect(result).toMatchSnapshot();
});
