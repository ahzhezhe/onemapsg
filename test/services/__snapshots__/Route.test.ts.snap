// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`route 1`] = `
{
  "phyroute": {
    "route_geometry": "uv}FkoxxR??",
    "route_instructions": [
      [
        "Straight",
        "CAVENAGH ROAD",
        0,
        "1.304274,103.841343",
        0,
        "0m",
        "North",
        "North",
        "driving",
        "Head North On Cavenagh Road",
      ],
      [
        "Straight",
        "CAVENAGH ROAD",
        0,
        "1.304274,103.841343",
        0,
        "0m",
        "North",
        "North",
        "driving",
        "You Have Arrived At Your Destination, Straight Ahead",
      ],
    ],
    "route_name": [
      "CAVENAGH ROAD",
    ],
    "route_summary": {
      "end_point": "CAVENAGH ROAD",
      "start_point": "CAVENAGH ROAD",
      "total_distance": 0,
      "total_time": 121,
    },
    "status": 0,
    "status_message": "Found route between points",
    "subtitle": "Shortest distance",
    "viaRoute": "CAVENAGH ROAD",
  },
  "route_geometry": "uv}FkoxxRIA",
  "route_instructions": [
    [
      "Left",
      "CAVENAGH ROAD",
      6,
      "1.304274,103.841343",
      1,
      "6m",
      "North",
      "North",
      "driving",
      "Head North On Cavenagh Road",
    ],
    [
      "Left",
      "CAVENAGH ROAD",
      0,
      "1.304322,103.841348",
      0,
      "0m",
      "North",
      "North",
      "driving",
      "You Have Arrived At Your Destination, On The Left",
    ],
  ],
  "route_name": [
    "CAVENAGH ROAD",
  ],
  "route_summary": {
    "end_point": "CAVENAGH ROAD",
    "start_point": "CAVENAGH ROAD",
    "total_distance": 6,
    "total_time": 1,
  },
  "status": 0,
  "status_message": "Found route between points",
  "subtitle": "Fastest route, assuming usual traffic",
  "viaRoute": "CAVENAGH ROAD",
}
`;
